### PR TITLE
Visualisation support for TetrahedralMeshes and ScalarTetrahedralMesh…

### PR DESCRIPTION
**Pull request by madsendennis**
Visualisation support for TetrahedralMeshes and ScalarTetrahedralMeshes. Dividing the previous pull-request into smaller chunks: (https://github.com/unibas-gravis/scalismo-ui/pull/38).

Screen Shot 2019-11-18 at 09 46 27

### CODE REVIEW

**Comment by clangguth** [comment]
Hi folks,

I'd like to look into this, but currently I'm struggling to find out exactly which scalismo branch or commit this depends on. How can I build `develop-c6d5f0abcb78661dd5f858df487326ed6009e693-SNAPSHOT` ?

Also, could you provide a code snippet to load (or produce) a `ScalarTetrahedralMeshField`?

Thanks!

**Comment by Andreas-Forster** [comment]
Hi Christoph,

This PR should not yet be merged.

A clean dependency of scalismo is in progress (ETA is the beginning of next week), then we can update this PR accordingly. The things to be merged are intended for "only" a _**Milestone 1**_ release, so they are not completely fixed, more in an advanced experimental stage. 

Further, the name of the class you asked about will also change. The creation then will be possible with something like:
```scala
object CreateScalarTetrahedralMeshField extends App {
  val mesh: TetrahedralMesh[_3D] = ???
  val data = mesh.pointSet.pointSequence.map(_ => 1.0)
  val svmf = ScalarVolumeMeshField(mesh,data)
}
```

If you prefer, we can after the PR is updated wait for your input before we merge it.

**Comment by clangguth** [comment]
Hi Andreas,

I'm fully aware that this is not ready for merging yet, and I'm myself definitely not going to merge anything, no worries :-) .

It's just that I was looking into the previous (discussion-only) PRs and had a few ideas for improvement that I wanted to look into -- very minor ones though, to be honest, I really like how it looks in general. But just as I was about to start, things were merged, and so I thought it'd make more sense to focus on the "newest version", so to say.

**Comment by marcelluethi** [comment]
@clangguth In the meantime I fixed the scalismo develop branch. You can do a publish-local of the last commit in the development branch and then use this as a dependency to scalismo-ui. With this it will work. Publishing a new scalismo version to bintray is planned for one of the next days.

File: src/main/scala/scalismo/ui/api/Views.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/api/Views.scala]: Is this g.peer.triangleMesh or tetrahedralMesh? If it is triangleMesh as a generic mesh term, we could rename it to meshes

File: src/main/scala/scalismo/ui/api/Views.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/api/Views.scala]: same as above

File: src/main/scala/scalismo/ui/api/Views.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/api/Views.scala]: same as above

File: src/main/scala/scalismo/ui/model/ScalarTetrahedralMeshFieldNode.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/model/ScalarTetrahedralMeshFieldNode.scala]: The line is too long. It should be broken into several lines

File: src/main/scala/scalismo/ui/rendering/actor/TetrahedralActor.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/rendering/actor/TetrahedralActor.scala:148]: Is this something you still want to fix?

File: src/main/scala/scalismo/ui/rendering/actor/mixin/ActorScalarGridRange.scala
Comment (marcelluethi) [inline src/main/scala/scalismo/ui/rendering/actor/mixin/ActorScalarGridRange.scala]: This does not seem to be needed. Can we remove it?

**Review by marcelluethi — COMMENTED** [review]
This looks good in general and also seems to work as intended. I think it can be merged, after the minor comments have been addressed.

File: src/main/scala/scalismo/ui/rendering/actor/TetrahedralActor.scala
Comment (madsendennis) [reply src/main/scala/scalismo/ui/rendering/actor/TetrahedralActor.scala:148]: This seems to be a direct copy from MeshActor where the FIXME originates from.

File: src/main/scala/scalismo/ui/rendering/actor/mixin/ActorScalarGridRange.scala
Comment (madsendennis) [reply src/main/scala/scalismo/ui/rendering/actor/mixin/ActorScalarGridRange.scala]: Yes. I'll remove it and push.

File: src/main/scala/scalismo/ui/api/Views.scala
Comment (madsendennis) [reply src/main/scala/scalismo/ui/api/Views.scala]: You are right. A bug seems to have snug in. It should have been g.peer.tetrahedralMeshes.

**Review by marcelluethi — APPROVED** [review]
Thanks for the changes. From my side I don't see any reasons why we should not merge this once we have a stable scalismo development branch.

**Comment by marcelluethi** [comment]
@madsendennis Updated the branch to use scalismo v0.18-M1. It's now ready to merge. 
Please don't forget to add all co-authors, when squashing the commits.

**Pull request merged**
